fix corepack & pnpm compatibility

xoldd opened the pull request:
**What kind of change does this PR introduce?**

bugfix

**Issue Number:**

Fixes #3145

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

Fixes the corepack and pnpm issues that were recently introduced because of some breaking changes in corepack. More information at [this](https://github.com/pnpm/pnpm/issues/9029) link.

**Does this PR introduce a breaking change?**

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Chores**
  - Streamlined development container setup for a simpler and more robust initialization process.
  - Updated deployment commands to simplify dependency installation.
  - Upgraded the runtime base environment and improved Node.js and package manager configurations for enhanced performance and compatibility.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

JaiPannu-IITI commented:
@xoldd, this initialize command don't work on Mac. That's why we changed it to be more compatible across operating systems.
Please check issue #3105.

xoldd commented:
> @xoldd, this initialize command don't work on Mac. That's why we changed it to be more compatible across operating systems. Please check issue #3105.

Yeah, deleted the associated commit. Thanks for letting me know.
